op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,258 @@
+package ws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/coder/websocket"
+
+	"github.com/ethereum-optimism/optimism/op-conductor/metrics"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	// reconnectDelay is the delay between reconnection attempts
+	reconnectDelay = 5 * time.Second
+	// pingInterval is how often to send pings to keep connections alive
+	pingInterval = 15 * time.Second
+	// pongTimeout is how long to wait for a pong response
+	pongTimeout = 10 * time.Second
+	// writeTimeout for all message writes
+	writeTimeout = 5 * time.Second
+	// send channel buffer size
+	sendChannelBufferSize = 256
+)
+
+// FlashblockHandler manages WebSocket connections for flashblocks
+type FlashblockHandler interface {
+	// Start initializes and starts the flashblocks handler
+	Start(ctx context.Context) error
+	// Stop closes any open WebSocket connections
+	Stop()
+	// BroadcastMessage sends a message to all connected WebSocket clients
+	BroadcastMessage(message []byte)
+}
+
+// Config contains configuration for the flashblocks handler
+type Config struct {
+	// WebsocketServerPort is the port to listen for WebSocket connections
+	WebsocketServerPort int
+	// RollupBoostWsURL is the URL of the rollup boost WebSocket
+	RollupBoostWsURL string
+}
+
+// Handler implements the FlashblockHandler interface
+type Handler struct {
+	cfg                 Config
+	log                 log.Logger
+	isLeaderFn          func(context.Context) bool
+	metrics             metrics.Metricer
+	rollupBoostConn     *websocket.Conn
+	rollupBoostCtx      context.Context
+	rollupBoostWsCancel context.CancelFunc
+	server              *http.Server
+	hub                 *Hub
+}
+
+// NewHandler creates a new flashblocks handler
+func NewHandler(cfg Config, log log.Logger, isLeaderFn func(context.Context) bool, m metrics.Metricer) (FlashblockHandler, error) {
+	// Validate configuration
+	if cfg.RollupBoostWsURL == "" || cfg.WebsocketServerPort <= 0 {
+		log.Error("rollup boost WebSocket URL or websocket server port not configured")
+		return nil, errors.New("rollup boost WebSocket URL or websocket server port not configured")
+	}
+
+	// Initialize the handler
+	handler := &Handler{
+		cfg:        cfg,
+		log:        log,
+		isLeaderFn: isLeaderFn,
+		metrics:    m,
+	}
+
+	// Try to establish initial connection to rollup boost WebSocket
+	maxConnectionAttempts := 5
+	var conn *websocket.Conn
+	var err error
+
+	for attempt := 1; attempt <= maxConnectionAttempts; attempt++ {
+		log.Info("attempting to connect to rollup boost WebSocket", "attempt", attempt, "url", cfg.RollupBoostWsURL)
+
+		// Create context with timeout for dialing
+		dialCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
for the connection, 30s is a little bit long, let's just do 5s for now. This is initialization phase, it's fine to fail if unable to connect